### ISSUE TITLE
ZCC_FILE_UPLOADER - Questions

### ISSUE THREAD

**Issue by axelmohnen** [abap2UI5/abap2UI5#550]
Hi,

1) How to include the 2UI5 JS - best practice?
I see in the demo app "Z2UI5_CL_DEMO_APP_075" that the "zcc_file_uploader_js" is set followed by a "SET_TIMER" method.
Do we really need this SET_TIMER. Is there another way?
```
    client->view_display( Z2UI5_cl_xml_view=>factory( client
         )->zcc_file_uploader_js(
         )->stringify( ) ).
```
` client->timer_set( event_finished = client->_event( `START` ) interval_ms = `0` ).`

2) I have the requirement to upload the file after selection immediately (property: uploadOnChange).
Is this covered with the "ZCC_FILE_UPLOADER" or do I have to created another "file upload" control?  

update:
3) The varibale binding for the file uploader doesn't work for nested views
Demo:
[ZAMOH_CL_DEMO_APP_065.txt](https://github.com/abap2UI5/abap2UI5/files/12659038/ZAMOH_CL_DEMO_APP_065.txt)

Thanks,
Kind regards
Axel

**Comment by choper725** [comment]
the use of the timer is just a way to load the custom control js code to the app, without it the js control is not working once you use the xml in the view

**Comment by oblomov-dev** [comment]
Hi @axelmohnen,

1) yes how @choper725 said, you need to define the custom control first before you can use it in an xml view, therefore we do two requests here and the first one immediately calls the backend again to load the xml view. theoretically we could also solve this in other ways, but this was in my tests and demos always the easiest way and where i do not need to make any changes to the normal abap2ui5 logic.
2) nice, having the possibility to upload directly is a good feature. i think merging both is a good idea. maybe with a new property "checkDirectUpload" and when this is set true in the backend, the extra button is not visible and the upload is directly done. i think the javascript code of both custom controls should be very similar. Feel free to merge it or i can do it when i am working with this control next time.
3) i need to check this.

best regards.

**Comment by axelmohnen** [comment]
FYI
I implemented the new property "checkDirectUpload" in order to merge both upload options into one control.
PR has been created.

**Comment by axelmohnen** [comment]
Issue 3 has been solved in the context of #553 .
Issue can be closed.
